buildtools/pipenv/pipenv.go

```diff
@@ -0,0 +1,156 @@
+package pipenv
+
+import (
+	"encoding/json"
+
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+// Pipenv enables python.go to use a custom Deps() function for testing.
+type Pipenv interface {
+	Deps() (graph.Deps, error)
+}
+
+// Executable allows a custom DataAccess function to be used by Deps().
+type Executable interface {
```

Comment:
This is overkill. I bet you'll find that you lose no flexibility or functionality if you remove this interface.

Reply:
This is what we were talking about earlier. This is the mockable dep that manages access to the executable build tool

Reply:
I chose to implement this as an interface to make it easier to extend in the future if need be. However, I do see how this could be too much. Are you envisioning something along the lines of the following?
```
type SystemPipenv struct {
	PipenvExec func(string) (string, error)
}
```